src/app/(maps)/components/FullScreenMap.tsx

```diff
@@ -1,11 +1,24 @@
 'use client'
-import { useEffect, useState } from 'react'
-import { GlobalMap } from '@/components/maps/GlobalMap'
+import { useCallback, useEffect, useState } from 'react'
+import { CameraInfo, GlobalMap } from '@/components/maps/GlobalMap'
+import { usePathname, useRouter, useSearchParams } from 'next/navigation'
 
 export const FullScreenMap: React.FC = () => {
   const [initialCenter, setInitialCenter] = useState<[number, number] | undefined>(undefined)
+  const [initialZoom, setInitialZoom] = useState<number | undefined>(undefined)
+  const router = useRouter()
+
+  const cameraParams = useCameraParams()
 
   useEffect(() => {
+    const initialStateFromUrl = cameraParams.fromUrl()
+
+    if (initialStateFromUrl != null) {
+      setInitialCenter([initialStateFromUrl.center.lng, initialStateFromUrl.center.lat])
+      setInitialZoom(initialStateFromUrl.zoom)
+      return
+    }
+
     getVisitorLocation().then((visitorLocation) => {
       if (visitorLocation != null) {
         setInitialCenter([visitorLocation.longitude, visitorLocation.latitude])
```

Comment:
Preview build: https://open-tacos-git-fork-d-k-lippert-feat-save-map-url-openbeta-dev.vercel.app/maps

In the preview build deployed on Vercel.com, `visitorLocation` will be non-null. I'm getting 'camera=1/0.00000/0.00000' and the map is no longer initialized to my approximate location.  Perhaps adding a check 'when initialStateFromUrl == null **and** visitorLocation != null  then set URL param to visitorLocation?

Reply:
Fixed ✅